test/Ocelot.UnitTests/DownstreamRouteFinder/DownstreamRouteFinderTests.cs

```diff
@@ -700,6 +826,36 @@ private void GivenTheUpstreamHttpMethodIs(string upstreamHttpMethod)
             _upstreamHttpMethod = upstreamHttpMethod;
         }
 
+        private void GivenUpstreamHeaderRoutingOptions()
+        {
+            var headers = new Dictionary<string, ICollection<string>>()
+            {
+                { "header", new[] { "value" }},
+            };
+            _upstreamHeaderRoutingOptions = new UpstreamHeaderRoutingOptions(headers, UpstreamHeaderRoutingTriggerMode.All);
+        }
+
+        private void GivenEmptyRequestHeaders()
+        {
+            _requestHeaders = new HeaderDictionary();
+        }

```

Comment:
Move this setup to the constructor, and remove all extra setup lines which are redundant!
Redundant call example is on line 685, and above.